text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,1301 @@
+
+```ts
+glue.ScalaSparkEtlJob(this, 'ScalaSparkEtlJob', {
+    script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+    className: 'com.example.HelloWorld',
+    role: iam.IRole,
+});
+
+glue.pySparkEtlJob(this, 'pySparkEtlJob', {
+    script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+    role: iam.IRole,
+});
+```
+
+Optionally, developers can override the glueVersion and add extra jars and a
+description:
+
+```ts
+glue.ScalaSparkEtlJob(this, 'ScalaSparkEtlJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+   extraJarsS3Url: [glue.Code.fromBucket('bucket-name', 'path-to-extra-scala-jar'),],
+   description: 'an example Scala Spark ETL job',
+   numberOfWorkers: 20,
+   workerType: glue.WorkerType.G8X,
+   timeout: cdk.Duration.minutes(15),
+   role: iam.IRole,
+});
+
+glue.pySparkEtlJob(this, 'pySparkEtlJob', {
+   jobType: glue.JobType.ETL,
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_9,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+   description: 'an example pySpark ETL job',
+   numberOfWorkers: 20,
+   workerType: glue.WorkerType.G8X,
+   timeout: cdk.Duration.minutes(15),
+   role: iam.IRole,
+});
+```
+
+Scala Spark ETL Job Property Interface:
+
+```ts
+ScalaSparkEtlJobProps{
+    /**
+     * Script Code Location (required)
+     * Script to run when the Glue Job executes. Can be uploaded
+     * from the local directory structure using fromAsset
+     * or referenced via S3 location using fromBucket
+     * */
+    script: glue.Code;
+
+    /**
+     * Class name (required for Scala)
+     * Package and class name for the entry point of Glue Job execution for
+     * Java scripts
+     * */
+    className: string;
+
+    /**
+     * Extra Jars S3 URL (optional)
+     * S3 URL where additional jar dependencies are located
+     */
+    extraJarsS3Url?: string[];
+
+    /**
+     * IAM Role (required)
+     * IAM Role to use for Glue Job execution
+     * Must be specified by the developer because the L2 doesn't have visibility
+     * into the actions the script(s) will take during the Job Execution
+     * */
+    role: iam.IRole;
+
+    /**
+     * Name of the Glue Job (optional)
+     * Developer-specified name of the Glue Job
+     * */
+    name?: string;
+
+    /**
+     * Description (optional)
+     * Developer-specified description of the Glue Job
+     * */
+    description?: string;
+
+    /**
+     * Number of Workers (optional)
+     * Number of workers for Glue to use during Job execution
+     * @default 10
+     * */
+    numberOrWorkers?: int;
+
+    /**
+     * Worker Type (optional)
+     * Type of Worker for Glue to use during Job execution
+     * Enum options: Standard, G.1X, G.2X, G.025X. G.4X, G.8X, Z.2X
+     * @default G.2X
+     * */
+    workerType?: glue.WorkerType;
```

Comment:
I'm just leaving the comment in this one place, but it stands for each instance here. Since these two are dependent on each other, it should be one field that takes in an enum like class. Something like `Worker.G_1X(200)` with the input being the number of workers.

Reply:
Worker type and number of workers are independent of each other which is why kept two different field. Customer can set number of workers and keep worker type as default (G.2X)